### ISSUE TITLE
Build break task failing

### ISSUE THREAD

**Issue by tlbonds2** [simondel/sonar-buildbreaker-vsts#10]
Hello, 
The build breaker task is all of a sudden failing with the following error.  Not sure if a change to the Azure DevOps API caused the following but wanted to reach out to see if you had any insight.  

Error:
##[section]Starting: Break build on quality gate failure
==============================================================================
Task         : SonarQube build breaker
Description  : Break the build when the quality gate fails.
Version      : 7.0.4
Author       : Simon de Lang
Help         : 
==============================================================================
Getting task report from file D:\AzDOAgents\Agent1\_work\53\.sonarqube\out\.sonar\report-task.txt
Retrieving analysisId from https://<OUR_URL>/sonar/api/ce/task?id=AWyMCyPRF-FKoHTXr7-L
##[error]Unknown result from previous http request. The body is: [object Object]. Statuscode is 200
Retrieving quality gate report from https://<OUR_URL>/sonar/api/qualitygates/project_status?analysisId=undefined
##[error]Something went wrong! Got statuscode 404 and error null
##[error]Unknown result from previous http request. The body is: [object Object]. Statuscode is 404
##[error]Unhandled: Cannot read property 'status' of undefined
##[section]Finishing: Break build on quality gate failure

**Comment by emi662002** [comment]
> Hello,
> The build breaker task is all of a sudden failing with the following error. Not sure if a change to the Azure DevOps API caused the following but wanted to reach out to see if you had any insight.
> 
> # Error:
> ##[section]Starting: Break build on quality gate failure
> # Task : SonarQube build breaker
> Description : Break the build when the quality gate fails.
> Version : 7.0.4
> Author : Simon de Lang
> Help :
> Getting task report from file D:\AzDOAgents\Agent1_work\53.sonarqube\out.sonar\report-task.txt
> Retrieving analysisId from https://<OUR_URL>/sonar/api/ce/task?id=AWyMCyPRF-FKoHTXr7-L
> ##[error]Unknown result from previous http request. The body is: [object Object]. Statuscode is 200
> Retrieving quality gate report from https://<OUR_URL>/sonar/api/qualitygates/project_status?analysisId=undefined
> ##[error]Something went wrong! Got statuscode 404 and error null
> ##[error]Unknown result from previous http request. The body is: [object Object]. Statuscode is 404
> ##[error]Unhandled: Cannot read property 'status' of undefined
> ##[section]Finishing: Break build on quality gate failure

Same here as of today, we are using Azure DevOps on premise so we have a stable api.

**Comment by JoopBananenbuiger** [comment]
Same issue here since this afternoon

**Comment by mtelle** [comment]
Same Here

**Comment by shoang-pfcu** [comment]
same here

**Comment by saikiranreddykandala** [comment]
Same Issue since Yesterday. It happens to All Projects.

**Comment by emi662002** [comment]
It this maybe related to an update to the sonarsube plugin for azure devops?

I've already opened a ticket there:
https://community.sonarsource.com/t/possible-error-publish-quality-gate-task-not-waiting-until-analysis-finishes/13148

Maybe check the commits done lately to see if the error is to be pinpointed, I didn't have the time so far

**Comment by dhiegobastos** [comment]
Same Issue since Yesterday. It happens to All Projects. (2)

**Comment by simondel** [comment]
Thanks for letting me know! I'll see if I can reproduce this.

**Comment by rwezowicz** [comment]
same here with multiple projects

**Comment by simondel** [comment]
Is this all on SonarQube or also some people with SonarCloud?

**Comment by rwezowicz** [comment]
> Is this all on SonarQube or also some people with SonarCloud?

We are using pure SonarQube

**Comment by shoang-pfcu** [comment]
sonarqube for me. and all projects.  I use TFS on premises.  So no azure

**Comment by tlbonds2** [comment]
sonarqube on-prem here!  Thanks for looking into it @simondel

**Comment by simondel** [comment]
May have been caused by https://github.com/SonarSource/sonar-scanner-vsts/releases/tag/sq-4.7.0
> Moved the generation of the report-task.txt file(s) in temp directory

That file is used in this extension.

_EDIT_ note to self: copy this logic https://github.com/SonarSource/sonar-scanner-vsts/blob/master/common/ts/sonarqube/TaskReport.ts#L43

**Comment by DeadlyDragon** [comment]
we have the same issue for on premise azure devops server 2019 and on-premise sonarqube as well.

**Comment by simondel** [comment]
# The issue has been resolved
Fixed in: 
* SonarCloud build breaker v1.0.11 (task version 1.0.7)
* SonarQube build breaker v7.0.7 (task version 7.0.5)

You can see which task version you run if you look at the logging of the task. The task version (not extension version) is shown at the start of the task.

## Updating
Azure DevOps Services users should be getting the new extension automatically. You can force an update by removing the task from your build agent folder.

Azure DevOps Server/TFS users should also get the new extension automatically, but Azure DevOps Server/TFS only runs the 'update extensions' job occasionally (once a day I think).

You can manually update the task by editing: 
```
MYAGENTFOLDER\_work\_tasks\sonarqube-buildbreaker_SOMEGUID\7.0.4\dist\common\checkQualityGateTask.js
```
(task folder name is different for sonarcloud, version to fix should be 1.0.6)

Replace the lines:
```js
const taskReportGlob = path.join('**', 'report-task.txt');
const taskReportGlobResult = tl.findMatch(tl.getVariable('Agent.BuildDirectory'), taskReportGlob);
```
with:
```js
const taskReportGlob = path.join('sonar', tl.getVariable('Build.BuildNumber'), '**', 'report-task.txt');
const taskReportGlobResult = tl.findMatch(tl.getVariable('Agent.TempDirectory'), taskReportGlob);
```

----------------------------------
I'm closing this issue for now as it has been resolved. If people still have issues, please let me know! Keep in mind that an automatic fix for on premise users may take up to a day.

**Comment by dhiegobastos** [comment]
@simondel now on version 7.0.5 I'm getting  the follow error
`Getting task report from file undefined
(node:670) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: path must be a string or Buffer`

Previously when the version 7.0.4 was ok, the output was
`The quality gate retrieved. The status of the quality gate is OK
The analysis has passed the quality gate: OK`

![Capturar](https://user-images.githubusercontent.com/7109329/63092137-18b2fa80-bf37-11e9-8602-7b55c5066ce1.PNG)

**Comment by tlbonds2** [comment]
Hey @simondel I am receiving the error as well.  Thoughts?
![image](https://user-images.githubusercontent.com/7121218/63093834-a3462a80-bf34-11e9-9c74-ffe58ec889ba.png)

**Comment by rwezowicz** [comment]
my builds are working correctly today

**Comment by cw-nvdael** [comment]
Yes, the plugin is working again. Tnx!

**Comment by simondel** [comment]
@dhiegobastos @tlbonds2 Could you try the latest version?

For SonarQube the new version is 8.* (https://github.com/simondel/sonar-buildbreaker-vsts/releases/tag/SonarQube-build-breaker-8.0.0)
For SonarCloud the new version is 2.* (https://github.com/simondel/sonar-buildbreaker-vsts/releases/tag/SonarCloud-build-breaker-2.0.2)

**Comment by dhiegobastos** [comment]
Hi @simondel
When I upgraded to version 8. * the pipeline didn't show those console errors again. But when I tried to simulate a Quality Gate fail scenario, the plug-in didn't break my build.

![Plugin](https://user-images.githubusercontent.com/7109329/63271149-e1fb1e00-c26f-11e9-9bb3-5e6ab9ea81c8.PNG)
![qualitygate](https://user-images.githubusercontent.com/7109329/63271153-e3c4e180-c26f-11e9-891f-634a5c34f6c3.PNG)
